src/software/SfM/main_openMVG2openMVS.cpp

```diff
@@ -34,65 +35,183 @@ using namespace openMVG::sfm;
-            if (imageIn.Contains(row, col))
+        for (int j = 0; j < height; ++j)
+            for (int i = 0; i < width; ++i)
             {
-                image_ud(j, i) = sampler(imageIn, row, col);
+                // compute coordinates with distortion
+                double row = (double)mapRow(j, i);
+                double col = (double)mapCol(j, i);
+                // pick pixel if it is in the image domain
+                if (imageIn.Contains(row, col))
+                {
+                    imageUd(j, i) = sampler(imageIn, row, col);
+                }
             }
-        }
+    }
+    else {
+        image::Sampler2d<image::SamplerNearest> nearestSampler;
+
+#ifdef OPENMVG_USE_OPENMP
+#pragma omp parallel for
+#endif
+        for (int j = 0; j < height; ++j)
+            for (int i = 0; i < width; ++i)
+            {
+                // compute coordinates with distortion
+                double row = (double)mapRow(j, i);
+                double col = (double)mapCol(j, i);
+                // pick pixel if it is in the image domain
+                if (imageIn.Contains(row, col))
+                {
+                    imageUd(j, i) = nearestSampler(imageIn, row, col);
+                }
+            }
+    }
 }
 
 template<typename RemapType>
 void CalcCameraDistortionMap(
     const openMVG::cameras::IntrinsicBase* cam,
-    RemapType& out_map_row,
-    RemapType& out_map_col
+    RemapType& outMapRow,
+    RemapType& outMapCol
 )
 {
     int width = cam->w();
     int height = cam->h();
 
-    out_map_row.resize(width, height);
-    out_map_col.resize(width, height);
+    outMapRow.resize(width, height);
+    outMapCol.resize(width, height);
 
 #ifdef OPENMVG_USE_OPENMP
     #pragma omp parallel for
 #endif
     for (int j = 0; j < height; j++)
         for (int i = 0; i < width; i++)
         {
-            const Vec2 undisto_pix(i, j);
+            const Vec2 undistoPix(i, j);
             // compute coordinates with distortion
-            const Vec2 disto_pix = cam->get_d_pixel(undisto_pix);
+            const Vec2 distoPix = cam->get_d_pixel(undistoPix);
 
-            out_map_row(j, i) = disto_pix[1];
-            out_map_col(j, i) = disto_pix[0];
+            outMapRow(j, i) = distoPix[1];
+            outMapCol(j, i) = distoPix[0];
         }
 }
 
+/// Check if the View have defined intrinsic
+bool IsIntrinsicDefined(const Intrinsics &intrinsics, const View *view)
+{
+  if (view == nullptr ) return false;
+  return (
+    view->id_intrinsic != UndefinedIndexT &&
+    intrinsics.find(view->id_intrinsic) != intrinsics.end());
+}
+
+bool exportToUndistortDepths(
+    const std::string& sInDir,
+    const SfM_Data& sfmData,
+    const std::string& sOutDir
+)
+{
+    if (sfmData.GetViews().empty())
+        return false;
+    int viewIdx = -1;
+    const auto &intrinsics = sfmData.GetIntrinsics();
+    for (const auto &v : sfmData.GetViews())
+    {
+        if (IsIntrinsicDefined(intrinsics, v.second.get()))
```

Comment:
Use SfMData::IsPoseAndIntrinsicDefined. Anyway, view is useless if pose is not defined.